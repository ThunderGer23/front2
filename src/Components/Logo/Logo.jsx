import PropTypes from 'prop-types'

export const Logo = ({Direccion}) =>{
    return(
        <>
        <div role='div' className="text-center mb-5">
            <img role='img' className="logo" src={Direccion} alt="" style={{width:'7rem'}}/>
        </div>
        </>
    )
}

Logo.propTypes={
    Direccion: PropTypes.string.isRequired
}