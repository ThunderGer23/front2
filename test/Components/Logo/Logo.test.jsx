import { render, screen } from "@testing-library/react";
import { Logo } from "../../../src/Components/Logo/Logo";

describe('Prueba en el componente Logo', () => { 
    const Direccion='../../../public/img/mq16.png'

    test('Esta prueba es para hacer match con el snapshot', () => { 
        const {container} = render(<Logo Direccion={Direccion}/>)//Renderiza el componente Link
        expect(container).toMatchSnapshot()//Toma foto
     })

    test('Esta prueba valida que el div exista y pertenezca a la clase correcta', () => { 
        render(<Logo Direccion={Direccion}/>)
        expect(screen.findByRole('div')).toBeTruthy()
        const {className} = screen.getByRole('div')//Se salva el tipo de la clase del div en className
        expect(className).toBe('text-center mb-5') 
     })

    test('Esta prueba valida que img exista y pertenezca a la clase correcta', () => { 
        render(<Logo Direccion={Direccion}/>)
        expect(screen.findByRole('img')).toBeTruthy()
        const {className} = screen.getByRole('img')
        expect(className).toBe('logo')
     })
 })