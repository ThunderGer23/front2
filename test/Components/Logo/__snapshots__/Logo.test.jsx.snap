// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente Logo Esta prueba es para hacer match con el snapshot 1`] = `
<div>
  <div
    class="text-center mb-5"
    role="div"
  >
    <img
      alt=""
      class="logo"
      role="img"
      src="../../../public/img/mq16.png"
      style="width: 7rem;"
    />
  </div>
</div>
`;
